eclair-core/src/main/scala/fr/acinq/eclair/payment/send/PaymentLifecycle.scala

```diff
@@ -130,9 +129,15 @@ class PaymentLifecycle(nodeParams: NodeParams, cfg: SendPaymentConfig, router: A
   }
 
   private def retry(failure: PaymentFailure, data: WaitingForComplete): FSM.State[PaymentLifecycle.State, PaymentLifecycle.Data] = {
-    val ignore1 = PaymentFailure.updateIgnored(failure, data.ignore)
-    router ! RouteRequest(nodeParams.nodeId, data.c.targetNodeId, data.c.finalPayload.amount, data.c.getMaxFee(nodeParams), data.c.assistedRoutes, ignore1, data.c.routeParams, paymentContext = Some(cfg.paymentContext))
-    goto(WAITING_FOR_ROUTE) using WaitingForRoute(data.c, data.failures :+ failure, ignore1)
+    data.c match {
+      case (sendPaymentToNode: SendPaymentToNode) =>
+        val ignore1 = PaymentFailure.updateIgnored(failure, data.ignore)
+        router ! RouteRequest(nodeParams.nodeId, data.c.targetNodeId, data.c.finalPayload.amount, sendPaymentToNode.getMaxFee(nodeParams), data.c.assistedRoutes, ignore1, sendPaymentToNode.routeParams, paymentContext = Some(cfg.paymentContext))
+        goto(WAITING_FOR_ROUTE) using WaitingForRoute(data.c, data.failures :+ failure, ignore1)
+      case (_: SendPaymentToRoute) =>
+        log.error("retrying SendPaymentToRoute")
+        stop(FSM.Normal)
```

Comment:
This should never happen as `maxAttempts` would be 1 and we wouldn't call `retry`.

Reply:
Then maybe the log message should be `unexpected retry during SendPaymentToRoute`?